client/src/components/Profiles.js

```diff
@@ -1,40 +1,83 @@
 import React, { useState, useEffect } from "react";
-import { Table } from "semantic-ui-react";
+import { Container, Message, Search, Table } from "semantic-ui-react";
 import { getProfiles } from "../api/profiles";
 
 const Profiles = () => {
   const [profiles, setProfiles] = useState([]);
+  const [searchInput, setSearchInput] = useState("");
+
+  const handleSearchChange = event => {
+    setSearchInput(event.target.value);
+  };
 
   useEffect(() => {
     getProfiles().then(response => {
-      setProfiles(response);
+      setProfiles(() => {
+        return !searchInput
+          ? response
+          : response.filter(
+              profile =>
+                profile.first_name
+                  .toLowerCase()
+                  .includes(searchInput.toLowerCase()) ||
+                profile.last_name
+                  .toLowerCase()
+                  .includes(searchInput.toLowerCase()) ||
+                profile.email
+                  .toLowerCase()
+                  .includes(searchInput.toLowerCase()) ||
+                (profile.address &&
+                  profile.address
+                    .toLowerCase()
+                    .includes(searchInput.toLowerCase())) ||
+                (profile.phone_number &&
+                  profile.phone_number
+                    .toLowerCase()
+                    .includes(searchInput.toLowerCase())) ||
+                (profile.occupation &&
+                  profile.occupation
+                    .toLowerCase()
+                    .includes(searchInput.toLowerCase()))
+            );
+      });
     });
-  }, []);
-  return !profiles.length ? (
-    <p>No Profiles!</p>
-  ) : (
-    <Table celled>
-      <Table.Header>
-        <Table.Row>
-          <Table.HeaderCell>#</Table.HeaderCell>
-          <Table.HeaderCell>Name</Table.HeaderCell>
-          <Table.HeaderCell>Phone</Table.HeaderCell>
-          <Table.HeaderCell>Email</Table.HeaderCell>
-        </Table.Row>
-      </Table.Header>
-      <Table.Body>
-        {profiles.map(profile => (
-          <Table.Row key={profile.id}>
-            <Table.Cell children={profile.id}></Table.Cell>
-            <Table.Cell
-              children={profile.first_name + " " + profile.last_name}
-            ></Table.Cell>
-            <Table.Cell children={profile.phone_number}></Table.Cell>
-            <Table.Cell children={profile.email}></Table.Cell>
-          </Table.Row>
-        ))}
-      </Table.Body>
-    </Table>
+  }, [searchInput]);
```

Comment:
This is loading the profiles every time there is a change to the `searchInput` field which causes unnecessary load on the server. You could instead write it so that the profiles are only loaded by `useEffect` when the component is first drawn and the filtering is done dynamically each time the component is redrawn (typically when the `searchInput` state changes). Also there is a lot of repeated code in the filtering so you could refactor that out into a separate function. I would write it something like this:

```
  useEffect(() => {
    getProfiles().then(response => {
      setProfiles(response);
    });
  }, []);

  const searchString = searchInput.toLowerCase();
  const isMatch = (field) => field.toLowerCase().includes(searchString);

  let filteredProfiles = profiles.filter(
    profile =>
      isMatch(profile.first_name) ||
      isMatch(profile.last_name) ||
      isMatch(profile.email) ||
      (profile.address && isMatch(profile.address)) ||
      (profile.phone_number && isMatch(profile.phone_number)) ||
      (profile.occupation && isMatch(profile.occupation))
  );
```

And you can then change the table to use `filteredProfiles` instead of `profiles`.